core/src/main/java/com/yahoo/imapnio/async/response/ImapResponseMapper.java

```diff
@@ -451,10 +465,99 @@ private SearchResult parseToSearchResult(@Nonnull final IMAPResponse[] ir) throw
                     while ((num = sr.readLong()) != -1) {
                         v.add(Long.valueOf(num));
                     }
+                    if (sr.readByte() == '(') {
+                        final String s = sr.readAtom();
+                        if (s.equalsIgnoreCase("MODSEQ")) {
+                            modSeq = sr.readLong();
+                        }
+                    }
+                }
+            }
+
+            return new SearchResult(v, modSeq);
+        }
+
+        /**
+         * Parses the responses from Store command and UID Store command to a @{code StoreResult} object.
+         *
+         * @param ir the list of responses from UID search command, the input responses array should contain the tagged/final one
+         * @return StoreResult object constructed based on the given IMAPResponse array,
+         * @throws ImapAsyncClientException when tagged response is not OK and NO or given response length is 0
+         */
+        @Nonnull
+        private StoreResult parseToStoreResult(@Nonnull final IMAPResponse[] ir) throws ImapAsyncClientException, IOException, ProtocolException {
+            if (ir.length < 1) {
+                throw new ImapAsyncClientException(FailureType.INVALID_INPUT);
+            }
+            final Response taggedResponse = ir[ir.length - 1];
+            if (!taggedResponse.isOK() && !taggedResponse.isNO()) {
+                throw new ImapAsyncClientException(FailureType.INVALID_INPUT);
+            }
+            final List<FetchResponse> fetchResponses = new ArrayList<>();
+            final List<Long> msgs = new ArrayList<>();
+
+            Long highestModSeq = null;
+
+            for (final IMAPResponse sr: ir) {
+                if (sr.isOK() || sr.isNO()) {
+                    sr.skipSpaces();
+                    if (sr.readByte() != (byte) L_BRACKET) {
+                        continue;
+                    }
+                    String s = sr.readAtom();
+                    if (s.equalsIgnoreCase("HIGHESTMODSEQ")) {
+                        highestModSeq = sr.readLong();
+                    } else if (s.equalsIgnoreCase("MODIFIED")) {
+                        s = sr.readAtom();
+                        s = s.replace("[", "");
+                        for (String str: s.split(",")) {
+                            msgs.add(Long.valueOf(str));
+                        }
+                    }
+                } else {
```

Comment:
Don't assume all untagged responses will be Fetch response, the server can potentially sent EXPUNGE or other responses, so be sure to parse before making a decision.

Reply:
Based on https://tools.ietf.org/html/rfc3501#section-6.4.6, Store will only have Fetch response. However, it is a good point that if they cast NOOP with EXPUNGE responses to StoreResults by mistake, there should be a way to be aware of that.

Reply:
Not necessarily, the server could sometime send untagged responses which is not a FETCH response. For example, Yahoo IMAP client will send an EXPUNGE response when storing a \Delete flag, with QRESYNC it will send a VANISHED response. 
So its better to identify the response type; inthis case FETCH, and then pasrse it or other response you should send back raw IMAPResponse